modules/calib3d/src/calibinit.cpp

```diff
@@ -451,11 +451,16 @@ int cvFindChessboardCorners( const void* arr, CvSize pattern_size,
     if( !out_corners )
         CV_Error( CV_StsNullPtr, "Null pointer to corners" );
 
-    if (img.channels() != 1)
+    if (img.channels() == 3)
     {
         cvtColor(img, img, COLOR_BGR2GRAY);
```

Comment:
COLOR_BGR2GRAY is enough. It works with both 3 and 4 channel images (like COLOR_BGRA2GRAY).
So this change is not needed at all.

Please eliminate unrelated commits too (`git rebase -i HEAD~2` should help).